src/tasks/minions/minigames/toaActivity.ts

```diff
@@ -0,0 +1,260 @@
+import { TeamLoot } from '../../../lib/simulation/TeamLoot';
+import { calcTOALoot, calculateXPFromRaid, toaOrnamentKits, toaPetTransmogItems } from '../../../lib/simulation/toa';
+import { TOAOptions } from '../../../lib/types/minions';
+import { formatOrdinal } from '../../../lib/util/formatOrdinal';
+import { handleTripFinish } from '../../../lib/util/handleTripFinish';
+import resolveItems from '../../../lib/util/resolveItems';
+import { updateBankSetting } from '../../../lib/util/updateBankSetting';
+import { userStatsUpdate } from '../../../mahoji/mahojiSettings';
+
+const purpleButNotAnnounced = resolveItems(['Dexterous prayer scroll', 'Arcane prayer scroll']);
+
+interface RaidResultUser {
+	points: number;
+	mUser: MUser;
+	deaths: number;
+	kc: number;
+}
+
+export const toaTask: MinionTask = {
+	type: 'TombsOfAmascut',
+	async run(data: TOAOptions) {
+		const { channelID, users, raidLevel, duration, leader, quantity, wipedRoom } = data;
+		const isSolo = users.length === 1;
+		const allUsers = await Promise.all(users.map(async u => mUserFetch(u[0])));
+		const leaderSoloUser = allUsers[0];
+
+		const previousCLs = allUsers.map(i => i.cl.clone());
+
+		// Increment all users attempts
+		await Promise.all(
+			allUsers.map(i =>
+				userStatsUpdate(i.id, () => ({
+					toa_attempts: {
+						increment: quantity
+					}
+				}))
+			)
+		);
+		if (wipedRoom) {
+			return handleTripFinish(
+				allUsers[0],
+				channelID,
+				`${allUsers.map(i => i.toString()).join(' ')} Your team wiped in the Tombs of Amascut!`,
+				undefined,
+				data,
+				null,
+				undefined
+			);
+		}
+
+		const totalLoot = new TeamLoot(toaCL);
+
+		const raidResults: Map<string, RaidResultUser> = new Map();
+		for (const user of allUsers) {
+			raidResults.set(user.id, {
+				mUser: user,
+				points: 0,
+				deaths: 0,
+				kc: await getMinigameScore(user.id, 'tombs_of_amascut')
+			});
+		}
+
+		let messages: string[] = [];
+
+		for (let x = 0; x < quantity; x++) {
+			const raidLoot = calcTOALoot({
+				users: users.map(i => {
+					const fullUser = allUsers.find(u => u.id === i[0])!;
+					return {
+						id: i[0],
+						points: i[1][x],
+						cl: fullUser.cl,
+						kc: raidResults.get(i[0])!.kc,
+						deaths: i[2][x]
+					};
+				}),
+				raidLevel
+			});
+			for (const [id, points, deaths] of users) {
+				const currentUser = raidResults.get(id)!;
+				currentUser.points += points[x];
+				currentUser.deaths += deaths[x].length;
+				currentUser.kc += 1;
+				raidResults.set(id, currentUser);
+			}
+			for (const [userID, userLoot] of raidLoot.teamLoot.entries()) {
+				totalLoot.add(userID, userLoot);
+			}
+			messages.push(...raidLoot.messages);
+		}
+		messages = uniqueArr(messages);
+		await Promise.all(allUsers.map(u => incrementMinigameScore(u.id, 'tombs_of_amascut', quantity)));
+
+		let resultMessage = isSolo
+			? `${leaderSoloUser}, your minion finished a Tombs of Amascut raid! Your KC is now ${await getMinigameScore(
```

Comment:
I think in a situation where a raid is solo, you could use the first(only) return value for previous promise.all(incrementMiniganesScore) for the new raid kc instead of having to await the getMinigameScore.